e2e-tests/trace_escrow_ledger_object/src/lib.rs

```diff
@@ -37,100 +55,140 @@ pub extern "C" fn finish() -> i32 {
+        #[cfg(target_arch = "wasm32")]
+        xrpl_wasm_stdlib::assert_eq!(ledger_entry_type, 117);
         let _ = trace_num("  LedgerEntryType:", ledger_entry_type as i64);
 
-        // Trace Field: CancelAfter
+        // Trace Field: CancelAfter (optional - require it for testing)
         let opt_cancel_after = current_escrow.get_cancel_after().unwrap();
-        if let Some(cancel_after) = opt_cancel_after {
-            assert_eq!(cancel_after, 545440232);
-            let _ = trace_num("  CancelAfter:", cancel_after as i64);
-        }
-
-        // Trace Field: Condition
-        let opt_condition = current_escrow.get_condition().unwrap();
-        if let Some(condition) = opt_condition {
-            assert_eq!(condition.0, EXPECTED_ESCROW_CONDITION);
-            let _ = trace_data("  Condition:", &condition.0, DataRepr::AsHex);
+        let cancel_after = opt_cancel_after.expect("CancelAfter should be set for testing");
+        let _ = trace_num("  CancelAfter:", cancel_after as i64);
+
+        // Trace Field: Condition (optional)
+        match current_escrow.get_condition() {
+            Ok(opt_condition) => {
+                if let Some(condition) = opt_condition {
+                    let _ = trace_num("  Condition length:", condition.len() as i64);
+                    let _ = trace_data(
+                        "  Condition (full hex):",
+                        condition.as_slice(),
+                        DataRepr::AsHex,
+                    );
+
+                    #[cfg(target_arch = "wasm32")]
+                    xrpl_wasm_stdlib::assert_eq!(
+                        condition.len(),
+                        EXPECTED_CONDITION.len(),
+                        "Condition length mismatch"
+                    );
+                    #[cfg(target_arch = "wasm32")]
+                    xrpl_wasm_stdlib::assert_eq!(
+                        condition.as_slice(),
+                        &EXPECTED_CONDITION[..],
+                        "Condition bytes mismatch"
+                    );
+                    let _ = trace("  ✓ Condition matches expected value");
+                } else {
+                    let _ = trace("  Condition: not present");
+                }
+            }
+            Err(e) => {
+                let _ = trace("  ERROR getting Condition");
+                let _ = trace_num("  error_code=", e as i64);
+                return e.code();
+            }
         }
 
         // Trace Field: Destination
         let destination = current_escrow.get_destination().unwrap();
-        const EXPECTED_DESTINATION: [u8; 20] = [
-            0x3E, 0x9D, 0x4A, 0x2B, 0x8A, 0xA0, 0x78, 0x0F, 0x68, 0x2D, 0x13, 0x6F, 0x7A, 0x56,
-            0xD6, 0x72, 0x4E, 0xF5, 0x37, 0x54,
-        ];
-        assert_eq!(destination.0, EXPECTED_DESTINATION);
+        // Destination is set in runTest.js (destWallet), just verify it's a valid AccountID
+        #[cfg(target_arch = "wasm32")]
```

Comment:
Why not make this a part of the macro instead of needing to specify it every time?

Reply:
Per our discussion today, I moved the macro into the `e2e-tests` folder so that nobody uses it accidentally in a real contract. See 63c3e7c760a1066ea5a610555e2d17f94e51ec1c